store/src/main/java/org/apache/rocketmq/store/CommitLog.java

```diff
@@ -964,9 +964,11 @@ public void run() {
             boolean result = false;
             for (int i = 0; i < RETRY_TIMES_OVER && !result; i++) {
```

Comment:
Consider `result = flush(0)`, the false result means there may be more data need to flush, so retry. Otherwise, the true result means all data are flushed, the broker can exit safely.

So if you change `result`, also please change this `for` loop.